### PR TITLE
Add unsigned 12 bit support

### PR DESCRIPTION
**Pull request by tfarago**

### CODE REVIEW

**Comment by tfarago** [comment]
Did I forget something?

**Comment by matze** [comment]
Looks good and very clean. I suppose there is a way to make use of SSE but for now I think it's alright, right?

**Comment by tfarago** [comment]
Exactly, unless there is a _lot_ of demand for processing data with 12 bpp I wouldn't optimize this.

File: ufo/ufo-buffer.c [hunk 1]
```diff
@@ -1215,6 +1216,31 @@ convert_data (UfoBufferPrivate *priv,
...
+        guchar first, second, third;
+        const guchar *src = (const guchar *) data;
+
+        if (n_pixels % 2) {
+            /* If there is an odd number of pixels we take care of the last two
+             * bytes forming the last pixel and process the rest in the loop. */
+            first = src[3 * n_pixels / 2 - 1];
+            second = src[3 * n_pixels / 2];
+            n_pixels -= 1;
+            dst[n_pixels - 1] = (gfloat) ((first << 4) | ((second & 0xF0) >> 4));
```
Comment (tfarago) [inline ufo/ufo-buffer.c]: This is of course wrong. Man, I didn't find example tiffs with odd number of pixels anywhere...

File: ufo/ufo-buffer.c [hunk 2]
```diff
@@ -1215,6 +1216,31 @@ convert_data (UfoBufferPrivate *priv,
...
+        guchar first, second, third;
+        const guchar *src = (const guchar *) data;
+
+        if (n_pixels % 2) {
+            /* If there is an odd number of pixels we take care of the last two
+             * bytes forming the last pixel and process the rest in the loop. */
+            first = src[3 * n_pixels / 2 - 1];
+            second = src[3 * n_pixels / 2];
+            dst[n_pixels - 1] = (gfloat) ((first << 4) | ((second & 0xF0) >> 4));
+            n_pixels -= 1;
```
Comment (tfarago) [inline ufo/ufo-buffer.c]: Now it's in the correct place.

**Comment by tfarago** [comment]
OK, done.

**Pull request merged**